Move SHOW and TERMINATE transactions to Ops Manual (#3034)
neo4j/docs-operations: modules/ROOT/pages/database-internals/show-and-terminate-transactions.adoc

:description: Information about the `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS` commands.
:test-skip: true
= List and terminate transactions

This page discusses how to use the `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS` to list and terminate currently running transactions.
For general information about the `SHOW` command, see the link:{neo4j-docs-base-uri}/cypher-manual/clauses/show/[Cypher Manual -> SHOW].

[[list-transactions]]
== List transactions

The `SHOW TRANSACTIONS` command displays currently running transactions across the instance.
To track transactions on remote database aliases, run `SHOW TRANSACTIONS` while connected to the remote database alias.

[[list-transactions-syntax]]

> === Syntax

Reviewer: no link to the how to read syntax descriptions page?

----
SHOW TRANSACTIONS YIELD *
----

.List transactions by transactionId
[source, cypher]
----
SHOW TRANSACTIONS "neo4j-transaction-3"
----

.Filter `SHOW TRANSACTIONS` with `WHERE`
[source, cypher]
----
SHOW TRANSACTIONS YIELD database, currentQuery
WHERE currentQuery contains 'Mark'
----

.Return renamed `SHOW TRANSACTIONS` columns
[source, cypher]
----
SHOW TRANSACTIONS
YIELD transactionId, elapsedTime, cpuTime, waitTime, idleTime,
  currentQueryElapsedTime, currentQueryCpuTime, currentQueryWaitTime, currentQueryIdleTime
RETURN
  transactionId AS txId,
  elapsedTime.milliseconds AS elapsedTimeMillis,
  cpuTime.milliseconds AS cpuTimeMillis,
  waitTime.milliseconds AS waitTimeMillis,
  idleTime.seconds AS idleTimeSeconds,
  currentQueryElapsedTime.milliseconds AS currentQueryElapsedTimeMillis,
  currentQueryCpuTime.milliseconds AS currentQueryCpuTimeMillis,
  currentQueryWaitTime.microseconds AS currentQueryWaitTimeMicros,
  currentQueryIdleTime.seconds AS currentQueryIdleTimeSeconds
----

[[terminate-transactions]]
== Terminate transactions

The `TERMINATE TRANSACTIONS` command is used to terminate running transactions by their IDs.

[[terminate-transactions-syntax]]

> === Syntax

Reviewer: also no link to syntax reading page

=== Return columns

All columns are returned by default for `TERMINATE TRANSACTIONS`.
As a result, appending `YIELD *` to `TERMINATE TRANSACTIONS` does not return any additional information.

`TERMINATE TRANSACTIONS` returns the following columns:

.Terminate transactions output
[options="header", cols="4,6,2"]
|===
| Column | Description | Type

m| transactionId
a| The transaction ID.
m| STRING

m| username
a| The username of the user executing the transaction.
m| STRING

m| message
a| The result of the `TERMINATE TRANSACTION` command as applied to this transaction.
m| STRING

|===

[[terminate-transactions-examples]]
=== Examples

.Terminate transactions by transaction ID
[source, cypher]
----
TERMINATE TRANSACTIONS "neo4j-transaction-1","neo4j-transaction-2"
----

.Filter `TERMINATE TRANSACTIONS` with `WHERE`
[source, cypher]
----
TERMINATE TRANSACTIONS "neo4j-transaction-1","neo4j-transaction-2"
YIELD transactionId, message
WHERE message <> "Transaction terminated."
----

[NOTE]
====
Unlike the `SHOW` commands, `TERMINATE TRANSACTIONS` does not allow `WHERE` without `YIELD`.
====

[[combine-transaction-commands]]
== Combine transaction commands

> `SHOW TRANSACTIONS` can be combined with `TERMINATE TRANSACTIONS` to identify and terminate transactions in the same statement.

Reviewer: might be nice to mention you can combine multiple of them (doesn't need to be a single one of each, you can have multiple show or multiple terminate or a mix of multiple of each...)

Reviewer: previously we had sentences like 
> The `SHOW TRANSACTIONS` command can be combined with multiple `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS`, see xref::clauses/transaction-clauses.adoc#query-combine-tx-commands[transaction commands combination].

up in the show and terminate sections and then
> When combining multiple commands ...

at the start of the note on needing YIELD and RETURN 🤷

[NOTE]
====
When combining `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS`, the `YIELD` clause is mandatory and must not be omitted.
In addition, `YIELD *` is not permitted.
Instead, the `YIELD` clause needs to explicitly list the yielded columns.

> The query must also end with a valid last clause (like a `RETURN` or an updating clause).

Reviewer: Feels odd to mention `updating clause` when talking only about combining show and terminate transaction
```suggestion
When combining `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS`,  the `YIELD` and `RETURN` clauses are mandatory and must not be omitted.
`YIELD` must explicitly list the yielded columns.
YIELD *` is not permitted.
```

All users can always view their own active transactions.

[[list-transactions-syntax]]
=== Syntax

For full details about the syntax descriptions, see xref:database-administration/syntax.adoc#administration-syntax-reading[Administration command syntax].

[options="header", width="100%", cols="1,5a"]
|===
| Action | Syntax

| List transactions on the current serve

a|
[source, syntax, role="noheader"]
----
SHOW TRANSACTION[S] [transaction-id[,...]]
[YIELD { * \| field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
[WHERE expression]
[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
----

[NOTE]
====
The format of `transaction-id` is `<databaseName>-transaction-<id>`.
Transaction IDs must be supplied as one or more comma-separated quoted `STRING` values, or as an expression resolving to a `STRING` or a `LIST<STRING>`.
As of Neo4j 2026.05, using Cypher 25, any transaction IDs that resolve to `null` are ignored instead of causing a type error exception.

> This is consistent with returning no results when the specified transaction ID does not exist.

Reviewer: Reneta suggested something like this on the show database PR
```suggestion
This behavior aligns with not returning anything when the given transaction ID does not exist.
```

----

[role=label--new-2026.05 label--cypher-25]
== Combine transaction commands with other Cypher commands

`SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS` can be combined with other Cypher commands to form a single query.

[NOTE]
====
When combining `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS`,  the `YIELD` and `RETURN` clauses are mandatory and must not be omitted.
`YIELD` must explicitly list the yielded columns.

> `YIELD *` is not permitted.

Reviewer: This is for when they are combined with other clauses and not just each other - that case is covered above. This have slightly different rules and those should be kept:
```suggestion
When combining `SHOW TRANSACTIONS` and `TERMINATE TRANSACTIONS` with other clauses, the `YIELD` clause is mandatory and must not be omitted.
`YIELD` must explicitly list the yielded columns.
`YIELD *` is not permitted.
```
